In [3]:
import pandas as pd
import numpy as np

In [5]:
df = pd.read_csv('insurance.csv')
df.head()

,age,sex,bmi,children,smoker,region,charges
0,19,female,27.900,0,yes,southwest,16884.92400
1,18,male,33.770,1,no,southeast,1725.55230
2,28,male,33.000,3,no,southeast,4449.46200
3,33,male,22.705,0,no,northwest,21984.47061
4,32,male,28.880,0,no,northwest,3866.85520


In [6]:
X = df.drop('charges', axis=1)
y = df['charges']

In [7]:
X = pd.get_dummies(X, drop_first=True)
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(
    X, y, test_size=0.2, random_state=42
)

# Linear Regression

In [12]:
from sklearn.linear_model import LinearRegression
lr = LinearRegression()
lr.fit(X_train, y_train)
y_pred_lr = lr.predict(X_test)

from sklearn.metrics import r2_score
print("Linear Regression R2 Score:", r2_score(y_test, y_pred_lr))

Linear Regression R2 Score: 0.7835929767120724


# Ridge Regression

In [11]:
from sklearn.linear_model import Ridge
ridge = Ridge(alpha=1.0)
ridge.fit(X_train, y_train)
y_pred_ridge = ridge.predict(X_test)

from sklearn.metrics import r2_score
print("Ridge Regression R2 Score:", r2_score(y_test, y_pred_ridge))

Ridge Regression R2 Score: 0.7832830144988215


# Random Forest

In [10]:
from sklearn.ensemble import RandomForestRegressor
rf = RandomForestRegressor(n_estimators=100, random_state=42)
rf.fit(X_train, y_train)
y_pred_rf = rf.predict(X_test)

from sklearn.metrics import r2_score
print("Random Forest R2 Score:", r2_score(y_test, y_pred_rf))

Random Forest R2 Score: 0.8651034329144947


# Conclusion

After performing Linear Regression, Ridge Regression, and Random Forest on the insurance dataset, we can conclude the accuracies as: <br>
&nbsp;&nbsp;&nbsp;&nbsp; Linear Regression : 0.7835929767120724<br>
&nbsp;&nbsp;&nbsp;&nbsp; Ridge Regression : 0.7832830144988215<br>
&nbsp;&nbsp;&nbsp;&nbsp; Random Forest : 0.8651034329144947<br>
<br>
Random Forest outperformed Linear and Ridge Regression, making it the best model for predicting insurance charges due to its ability to capture complex patterns in the data.